pyk/src/pyk/konvert/_ast_to_kast.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+import logging
+from functools import singledispatch
+from typing import TYPE_CHECKING, cast
+
+from ..kast.att import EMPTY_ATT
+from ..kast.outer import KAtt, KDefinition, KFlatModule, KImport, KRequire, KSentence
+from ..kast.outer_syntax import Att, Definition, Import, Module, Require  # noqa: TC002
+
+if TYPE_CHECKING:
+    from typing import Any, Final
+
+    from ..kast.outer import KAst
+    from ..kast.outer_syntax import AST
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+@singledispatch
```

Comment:
Nice.

Reply:
This is a trick I didn't know about; very cool

Reply:
I had to massage the type checker a bit to accept this in a few places, but it does make for some convenient calls like the one here:

https://github.com/runtimeverification/k/blob/7cbf5551b5aca2e7a19a231611c8ceeb08bc5f0a/pyk/src/pyk/konvert/_ast_to_kast.py#L35

Reply:
I guess it saves you a big pile of `isinstance` checks which is nice

Reply:
Came up with a better solution than the KSentence cast above: a10e68aeed9ce8fd4d3cfbb7cfa59bb1089980e8